src/models/team.ts

```diff
@@ -0,0 +1,249 @@
+			);
+		}
+
+		return pine.post({
+			resource: 'team',
+			body: {
+				name,
+				belongs_to__organization: orgId,
+			},
+		});
+	};
+
+	/**
+	 * @summary Get all Teams of a specific Organization
+	 * @name getAllByOrganization
+	 * @public
+	 * @function
+	 * @memberof balena.models.team
+	 *
+	 * @param {Number} organizationSlugOrId - Required: the organization slug or id the team is part of.
+	 * @param {Object} [options={}] - extra pine options to use
+	 *
+	 * @fulfil {Object[]} - Teams
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.team.getAllByOrganization(123).then(function(teams) {
+	 * 	console.log(teams);
+	 * });
+	 *
+	 * @example
+	 * balena.models.team.getAllByOrganization('MyOrganizationHandle').then(function(teams) {
+	 * 	console.log(teams);
+	 * });
+	 */
+	const getAllByOrganization = function (
+		organizationSlugOrId: string | number,
+		options: BalenaSdk.PineOptions<BalenaSdk.Team> = {},
+	): Promise<BalenaSdk.Team[]> {
+		return pine.get({
+			resource: 'team',
+			options: mergePineOptions(
+				{
+					$filter: {
+						belongs_to__organization: isId(organizationSlugOrId)
+							? organizationSlugOrId
+							: { handle: organizationSlugOrId },
+					},
+				},
+				options,
+			),
+		});
+	};
+
+	/**
+	 * @summary Get a single Team
+	 * @name get
+	 * @public
+	 * @function
+	 * @memberof balena.models.team
+	 *
+	 * @param {Number} teamId - team id (number).
+	 * @param {Object} [options={}] - extra pine options to use
+	 *
+	 * @fulfil {Object} - Team
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.team.get(123).then(function(team) {
+	 * 	console.log(team);
+	 * });
+	 */
+	const get = async function (
+		teamId: number,
+		options: BalenaSdk.PineOptions<BalenaSdk.Team> = {},
+	): Promise<BalenaSdk.Team> {
+		if (teamId == null) {
+			throw new errors.BalenaInvalidParameterError('id', teamId);
+		}
+
+		const team = await pine.get({
+			resource: 'team',
+			id: teamId,
+			options,
+		});
+		if (team == null) {
+			throw new Error(`Team not found: ${teamId}`);
+		}
+		return team;
+	};
+
+	/**
+	 * @summary Rename Team
+	 * @name rename
+	 * @public
+	 * @function
+	 * @memberof balena.models.team
+	 *
+	 * @param {Number} teamId - team id (number)
+	 * @param {String} newName - new team name (string)
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.team.rename(123, 'MyNewTeamName');
+	 */
+	const rename = async function (
+		teamId: number,
+		newTeamName: string,
+	): Promise<void> {
+		const team = await get(teamId, {
+			$select: 'id',
+			$expand: { belongs_to__organization: { $select: 'id' } },
+		});
+		console.log(team);
+		if (team == null) {
+			throw new Error(`Team not found: ${teamId}`);
+		}
+
+		const orgId =
+			Array.isArray(team.belongs_to__organization) &&
+			team.belongs_to__organization[0] &&
+			'id' in team.belongs_to__organization[0]
+				? team.belongs_to__organization[0].id
+				: undefined;
```

Comment:
If we do
```
		const teamOptions = {
			$select: ['id', 'belongs_to__organization'],
		} satisfies PineOptions<Team>;
		const team = (await get(teamId, teamOptions)) as PineTypedResult<Team, typeof teamOptions>;
```
then this will be fully typed and we should be able to do just:
```suggestion
		const orgId = team.belongs_to__organization.__id;
```